lib/xlibc/include/assert.h

```diff
@@ -0,0 +1,41 @@
+#ifndef _XLIBC_ASSERT_H
+#define _XLIBC_ASSERT_H
+
+/* based on _PDCLIB_symbol2value from PDCLIB */
+#ifndef _XLIBC_symbol2value
+#define _XLIBC_symbol2value(x) #x
+#endif
+
+/* based on _PDCLIB_symbol2string from PDCLIB */
+#ifndef _XLIBC_symbol2string
+#define _XLIBC_symbol2string(x) _XLIBC_symbol2value(x)
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#undef assert
+#ifdef NDEBUG
+    #define assert(ignore) ((void)0)
```

Comment:
```
void foo() {}
```

and later:

```
foo() // Missing ; here!
assert(z == 10);
```

causes a syntax error in my host clang, but not in nxdk.

Error in my clang is:
```
test.c:27:5: error: called object type 'void' is not a function or function
      pointer
    assert(z == 10);
    ^
/usr/include/assert.h:50:24: note: expanded from macro 'assert'
# define assert(expr)           (__ASSERT_VOID_CAST (0))
                                ^
1 error generated.
```

It is independent of NDEBUG in my host.